include/spdlog/details/fmt_helper.h

```diff
@@ -23,6 +23,20 @@ inline spdlog::string_view_t to_string_view(const memory_buf_t &buf) SPDLOG_NOEX
     return spdlog::string_view_t{buf.data(), buf.size()};
 }
 
+#ifdef SPDLOG_USE_STD_FORMAT
+template<typename T>
+std::basic_string<T> to_string(const std::basic_string<T> &buf)
```

Comment:
maybe could be optimize by changing the signature to 
```std::basic_string<T> &. to_string(const std::basic_string<T> &buf)```

Reply:
We don't actually benefit much from this specific form. All usages of `to_string` are to return by value or push_back, so we'll end up doing a copy anyways. A better form would be `std::basic_string<T> to_string(std::basic_string<T> &&buf)`, as this wouldn't copy. I'll apply that.